test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeArrayTests.java

```diff
@@ -0,0 +1,767 @@
+	 */
+	@Test(priority=1,invocationCount=2)
+	static public void testValueTypeArrayAssignments() throws Throwable {
+		Object[][] testArrays = new Object[][] {new Object[2], new SomeIface[2], new PointV[2], new PointFV![2]};
+		int[] kinds = {OBJ_TYPE, IFACE_TYPE, VAL_TYPE, FLATTENABLE_TYPE};
+		Object[] vals = new Object[] {null, bogusIfaceObj, new PointV(1.0, 2.0), new PointFV(3.0, 4.0)};
+
+		for (int i = 0; i < testArrays.length; i++) {
+			Object[] testArray = testArrays[i];
+
+			for (int j = 0; j < kinds.length; j++) {
+				int staticArrayKind = kinds[j];
+
+				for (int k = 0; k < kinds.length; k++) {
+					int staticValueKind = kinds[k];
+					// runTest's parameters are of type Object[] and Object.  It ultimately dispatches to an assign
+					// method whose parameters have more specific static types.  This condition filters out the
+					// combinations of static types that aren't permitted from the point of view of the Java language.
+					//
+					// Cases:
+					//  - the two types are the same  (staticArrayKind == staticValueKind)
+					//  OR
+					//    - the type of the array is Object[] or SomeIface[]  (staticArrayKind < VAL_TYPE)
+					//      AND
+					//    - the type of the array is less specific than that of the source (staticArrayKind < staticValueKind)
+					//
+					if (staticArrayKind == staticValueKind
+						|| (staticArrayKind < staticValueKind && staticArrayKind < VAL_TYPE)) {
+						runTest(testArrays[i], nullObj, staticArrayKind, staticValueKind);
+						runTest(testArrays[i], bogusIfaceObj, staticArrayKind, staticValueKind);
+						runTest(testArrays[i], pointVal, staticArrayKind, staticValueKind);
+						runTest(testArrays[i], pointFlattenableVal, staticArrayKind, staticValueKind);
+					}
+				}
+			}
+		}
+	}
+
+	static value class SomeFlattenableClassWithDoubleField {
+		public double d;
+
+		public implicit SomeFlattenableClassWithDoubleField();
+
+		SomeFlattenableClassWithDoubleField(double x) {
+			this.d = x;
+		}
+	}
+
+	static value class SomeFlattenableClassWithFloatField {
+		public float f;
+
+		public implicit SomeFlattenableClassWithFloatField();
+
+		SomeFlattenableClassWithFloatField(float x) {
+			this.f = x;
+		}
+	}
+
+	static value class SomeFlattenableClassWithLongField {
+		public long l;
+
+		public implicit SomeFlattenableClassWithLongField();
+
+		SomeFlattenableClassWithLongField(long x) {
+			this.l = x;
+		}
+	}
+
+	static class SomeIdentityClassWithDoubleField{
+		public double d;
+
+		SomeIdentityClassWithDoubleField(double x) {
+			this.d = x;
+		}
+	}
+
+	static class SomeIdentityClassWithFloatField{
+		public float f;
+
+		SomeIdentityClassWithFloatField(float x) {
+			this.f = x;
+		}
+	}
+
+	static class SomeIdentityClassWithLongField{
+		public long l;
+
+		SomeIdentityClassWithLongField(long x) {
+			this.l = x;
+		}
+	}
+
+   interface SomeInterface1WithSingleImplementer {}
+
+   interface SomeInterface2WithSingleImplementer {}
```

Comment:
It looks like these two interface declarations are the last two lines that are still indented with spaces.